### DISCUSSION TITLE
Connect two jamulus instances on two different continents

### DISCUSSION THREAD

**Discussion by openaspace** [jamulussoftware/jamulus#2182] [Technical Support & Help]
Hello.
I have a music ensemble in United States and one in Europe that need to play together.. my question is:
To optimize the interplay of the two ensemble, can I create two jamulus servers, one in United States and one in Europe.. connecting together the two server?  instead to create only one server in one continent and have big latency from the more distant musicians?

Thank you.

**Comment by dcorson-ticino-com** [comment]
That is not possible and it wouldn't change much as the latency between the groups would remain.
I have done a lot of playing "across the pond" recently trying Jamulus and Kamkazam.
The results: the lower latency provided by Jamkazam's peer-to-peer architecture does not help as there is no synchronisation.  It is much harder to play together using Jamkazam with 45ms latency than using Jamulus with 90ms.  Using Jamulus it is possible to anticipate and synchronise, it takes a while to get used to, but works.  With Jamkazam as there is no synchronisation everyone hears something different, playing together is only possible with very low latencies.
The choice of repertoire is also very important.  Keep to things that are not rhythmically complicated and you will have better chances of getting a good result.  To get a result you will also need players who are flexible enough to try out different things and not just play as always and complain that playing over the Internet is not like playing together in the same room.  Of course it isn't, you will need to change your habits some.
I wish you much success,
Don

---EDIT---
And rule Nr1 using Jamulus to be able to play together: Everyone listens what is coming back from the server, never listen to your own direct input.

**Reply by openaspace** [reply]
Thank you, but i need to optimize the two groups, if I can have 5 musicians in sync in USA and five in sync in Europe, connected together by the server I think could be a nice solution instead to have 5 of 10 musicians with a latency of 120ms..

Obviously if the server technology don't support the server link... There isn't much to do .. but a professional server link will have surely less latency.

Per to peer it's not a considerable solution. For example I use sonobus only for 2 musicians with multi channels remote recording.

Thank you.

**Reply by dcorson-ticino-com** [reply]
Considering that the speed of light in a fiber optic cable is about 5ns/meter and there are also repeaters, switches etc between the end servers I don't think that even the very best connections are much better than what we are seeing.
Knowing that it is not easy to get around the laws of physics, I think it is really incredible the service we are getting these days.

**Reply by chrisrimple** [reply]
Having the musicians in a country in sync together, but then having that group of musicians out-of-sync with those in another country (due to latency), will result in frustration and disaster.  The solution is to configure a Jamulus cloud server very close to one of the groups, and the other group learns to play by adapting to the latency.  This happens on the [WorldJam](http://worldjam.vip) ([videos](https://www.youtube.com/c/WorldJam/videos)) all the time - there are frequently songs played by musicians in Europe and the USA East Coast, using a server in London or New Jersey.  [Here's an example from last week in which most performers are in Europe on a London server, while the backing vocalist is in the USA](https://www.youtube.com/watch?v=I6PGuvCcgos&t=4408s), and there are many more.  Personally, I've played on WorldJam from my home in Seattle (USA West Coast) with musicians in Toronto Canada, where they had about 80ms of latency, and you wouldn't know that we were in different cities/countries - among the WorldJam performers, it's not uncommon.

**Reply by openaspace** [reply]
Thank you to all.
I already know the psychics concepts and how to play online.

Simply I need to technically link two servers.. If this it's not possible I will find another solutions using network technics.

Thank you to all.

**Reply by gene96817** [reply]
It is very tempting to create some "local" synchronization. If you map out the pairwise latency between musicians, as they would hear each other, you will discover the local clusters don't solve the problem. Special cases can be created where it is assumed that some musicians don't have to hear each other; however, then you don't have a real-time ensemble.

Having said that, if you have a few servers with special (expensive) low latency paths, then what helps is using the nearby server as a router to access a special low latency path. This works if you assume ordinary musicians use commodity Internet links with higher latency. If you don't set up special very low latency server-to-server links, creating local clusters of musicians won't provide any advantage to manage latency.

**Reply by openaspace** [reply]
Really Thanks to all.
Yes this is the approach, group the two ensemble in the two continents in different near country's.. therefore join the two servers. Each ensemble will play non at the same time, and the final output of the two server, steamed on Amazon aws cdn.

**Comment by ghost** [comment]
This is feasible:
- two separate groups on their local servers for low local latency.
- the lead group leads but does not hear the lag group, the lag group follows by hearing the lead group.
- a client located near the following (lag) server connects to the following server. another client located near the following server (and on a different computer than the other client) connects to the leading server.
- the audio of the two clients are connected by external cables between their two computers.
- it is now possible for the lag group clients to hear the lead group.

**Reply by gene96817** [reply]
> the lead group leads but does not hear the lag group, the lag group follows by hearing the lead group.

For this requirement, it would be easier to just provide a recording to the lag group.
Using a recording is also much more convenient if the lag group is more than two time zones away from the lead group. (Two connected servers in Europe and North America would be 4-6 hours apart. Europe and Africa could be in the same time zone.)
To each their own. If I was in the lag group, this is as interesting as playing with an Internet radio station.

**Comment by chrisrimple** [comment]
Contact [Tord Karlin](https://www.facebook.com/profile.php?id=100011670519922) and/or [Laura Mardones](https://www.facebook.com/laura.elena.mardones) - you may also reach her via https://musicbridge.world/.  Tord has led some successful experiments in chaining servers to allow 100+ singers to perform together, and Laura has been closely involved.

**Comment by pljones** [comment]
Have you watched WorldJam?  You'll see people playing from Russia, Europe and the Eastern and middle states of the US and Canada happily.  They use one server.

How?

It takes _practice_.

If you use one server - _and listen only to that_ - then you can learn how to do it, too.

If you try any other solution, it won't work.  It won't sync.  It'll sound terrible and you'll be disappointed with having wasted the time.
